specs/spaces-implementation.md

```diff
@@ -0,0 +1,51 @@
+Technical specification is [here](./spaces-specification.md)
+# Spaces implementation
+
+## API
+
+`openBox` will have a new option:
+```js
+const opts = {
+  spaces: {
+    <name1>: 'public',
+    <name2>: 'private',
+    <name3>: 'both'
+}
+```
+Passing this option you will open the given space(s) at the same time as opening the main space.
+
+## `box.spaces.open(<name>, <mode>)`
+
+Here *name* is the name of the space and *mode* is `'public'`, `'private'`, or `'both'`
+
+The new space will be available as a keyValueStore as follows: 
+
+`box.spaces.<name>.public`
+
+`box.spaces.<name>.private`
+
+## Key management
+
+We add a new class to 3box-js called Keyring. This class can hold any number of (signing-key-pair, symmetric encryption-key) tuples. It also provides methods to symmetrically encrypt data. We can base it loosely on the keyring in muport-core, but lets use [https://www.npmjs.com/package/hdkey](https://www.npmjs.com/package/hdkey) instead of ethereumjs-wallet (it uses hdkey internally, we only need that functionallity).
```

Comment:
oh, nice ethers.js also supports the creation of bip39 mnemonics so we can remove the `bip39` dependancy as well!

Reply:
Here is the documentation for HD node: https://docs.ethers.io/ethers.js/html/api-advanced.html#hdnode

Reply:
@oed OK let's use ethers.js!

Reply:
@christianlundkvist This could be the same keyring module as we want to use in the muPort refactor. That would be great because it would reduce the amount of code that we need to write.